core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -126,29 +132,8 @@ class ControllerContext {
   }
 
   def partitionFullReplicaAssignment(topicPartition: TopicPartition): ReplicaAssignment = {
-    partitionAssignments.getOrElse(topicPartition.topic, mutable.Map.empty).get(topicPartition.partition) match {
-      case Some(partitionAssignment) => partitionAssignment
-      case None => ReplicaAssignment(Seq(), Seq(), Seq())
-    }
-  }
-
-  def updatePartitionReplicaAssignment(topicPartition: TopicPartition, newReplicas: Seq[Int]): Unit = {
```

Comment:
The bulk of this change is the removal of this method and the changes related to this removal.